ui/raidboss/data/03-hw/alliance/dun_scaith.js

```diff
@@ -145,62 +152,94 @@
       netRegexJa: NetRegexes.startsUsing({ id: '1C98', source: 'フェルディア・ホロー' }),
       netRegexCn: NetRegexes.startsUsing({ id: '1C98', source: '虚空弗迪亚' }),
       netRegexKo: NetRegexes.startsUsing({ id: '1C98', source: '공허의 페르디아' }),
+      condition: Conditions.caresAboutMagical(),
       response: Responses.tankBuster(),
     },
+    {
+      // 5510 is Wailing Atomos, 5511 is Cursing Atomos.
+      // Sometimes it will happen that Aether/Chakrams will start casting before
+      // the addedCombatant line that contains the Atomos.
+      // When this happens, a simple startsCasting trigger will silently fail.
+      // To avoid this, we store the IDs of Atomos for later comparison.
+      id: 'Dun Scaith Atomos Setup',
+      netRegex: NetRegexes.addedCombatantFull({ npcNameId: ['5510', '5511'] }),
+      run: function(data, matches) {
+        data.cursing = data.cursing || [];
+        data.wailing = data.wailing || [];
+        matches.npcNameId == '5510' ? data.wailing.push(matches.id) : data.cursing.push(matches.id);
+      },
+    },
     {
       // Wailing Atomos is blue, Cursed Atomos is yellow.
       // 1C9F:Aether is the circle AoE, 1CA0:Aetherial Chakram is the donut AoE
-      id: 'Dun Scaith Blue Atomos',
-      netRegex: NetRegexes.startsUsing({ id: ['1C9F', '1CA0'], target: 'Wailing Atomos' }),
-      netRegexDe: NetRegexes.startsUsing({ id: ['1C9F', '1CA0'], target: 'Heul-Atomos' }),
-      netRegexFr: NetRegexes.startsUsing({ id: ['1C9F', '1CA0'], target: 'Gueule Gémissante' }),
-      netRegexJa: NetRegexes.startsUsing({ id: ['1C9F', '1CA0'], target: '虚声のアトモス' }),
-      netRegexCn: NetRegexes.startsUsing({ id: ['1C9F', '1CA0'], target: '虚声的阿托莫斯' }),
-      netRegexKo: NetRegexes.startsUsing({ id: ['1C9F', '1CA0'], target: '허성의 아토모스' }),
-      alertText: function(data, matches) {
-        if (matches.id == '1C9F') {
-          return {
-            en: 'Avoid Untethered Blue',
-            de: 'Weiche dem nicht verbundenen blauem Atomos aus',
-            fr: 'Evitez Gueule bleue non-liée',
-            cn: '远离蓝色小怪',
-          };
-        }
-        if (matches.id == '1CA0') {
-          return {
-            en: 'Go to Untethered Blue',
-            de: 'Gehe zu dem nicht verbundenen blauem Atomos',
-            fr: 'Allez vers la Gueule bleue non-liée',
-            cn: '靠近蓝色小怪',
-          };
-        }
+      id: 'Dun Scaith Atomos Compile',
+      netRegex: NetRegexes.startsUsing({ id: ['1C9F', '1CA0'] }),
+      delaySeconds: .5,
+      run: function(data, matches) {
+        data.atomos = data.atomos || 0;
```

Comment:
I think my inclination when I see a pattern like this that's mapping some more complicated set of data into and out of an integer is to suggest to write it out explicitly.  What do you think about something more along these lines:

```
data.sphere = data.sphere || [];
data.donut = data.donut || [];
const target = data.wailing.includes(matches.targetId) ? 'wailing' : 'cursing';
if (matches.id === '1C9F')
  data.sphere.push(target);
else
  data.donut.push(target);
```

And then later on you could enumerate the possibilities:
```
if (data.donut.length === 2)
  // on any
else if (data.sphere.length === 2)
  // avoid all
else if (data.donut.length === 1)
  if (data.donut[0] === 'wailing')
    // on wailing
  else
    // on cursing  
else // remaining case: data.sphere.length === 1 && data.donut.length === 0
  if (data.sphere[0] === 'wailing')
    // avoid wailing
  else
    // avoid cursing

```

Reply:
Reasonable. I was more trying to avoid the awful IF tower, and to keep the response localizations all in one place together, but this is definitely just as readable.

(I suppose this is as good a place to ask, are we moving to `===` strict comparison generally now? I would prefer that, but to date the repository seems to use `==` loose.)

Reply:
The if tower isn't great, but it's at least much more readable.  I'm not sure in this case that there's any way to avoid it.

(Re: comparisons.  Yeah, past behavior was mostly just "we were not good JavaScript programmers, sorry." I think my inclination is that we use strict where we can, especially in new code.)